distributed/worker_state_machine.py

```diff
@@ -3656,9 +3656,11 @@ def _finish_async_instruction(
             stim = task.result()
         except asyncio.CancelledError:
             # This should exclusively happen in Worker.close()
+            logger.warning(f"Async instruction for {task} ended with CancelledError")
             return
         except BaseException:  # pragma: nocover
-            logger.exception("async instruction handlers should never raise!")
+            # This should never happen
+            logger.exception(f"Unhandled exception in async instruction for {task}")
```

Comment:
Not intended as  blocking comment, but I could see it being useful if we included the type of the exception and possibly the exception message here.

Reply:
`logger.exception` automatically does that